c10/core/Stream.h

```diff
@@ -92,6 +93,14 @@ class Stream final {
   DeviceIndex device_index() const noexcept { return device_.index(); }
   StreamId id() const noexcept { return id_; }
 
+  // Enqueues a wait instruction in the stream's work queue.
+  // This instruction is a no-op unless the event is marked
+  // for recording. In that case the stream stops processing
+  // until the event is recorded.
+  void wait(const impl::EventInterface& event) const {
```

Comment:
I'm not convinced `EventInterface` is necessary here. Can't you just make this templated, so that it works on `Event` or `InlineEvent`?

Reply:
Yes we can. I'll find a different home for the enum.